### PR TITLE
Boas 805 api testing setup

### PR DESCRIPTION
**Pull request by ob-pins**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

**Review by mayya-bond-kc — APPROVED** [review]
Looking good!

**Review by LetsMakeItTechnical — APPROVED** [review]

File: apps/api-testing/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+## Introduction
+This framework uses [Supertest](https://github.com/visionmedia/supertest), [Mocha](https://mochajs.org/), and [Ajv](https://ajv.js.org/) for schema validation. 
+
+## Getting started
+To get started with the API test framework, follow these steps:
+1. Install the required dependencies by running `npm install` from the root of the Back Office project.
+2. Run the shell script by typing `./test.sh\`. or run `npm run api:test` from the root for the Back Office project.
+5. Create a `.env` file in `apps/api-testing` directory.
+6. Add a `BASE_URL` variable to the `.env` file. The value should be the URL of the API being tested (e.g. \`BASE_URL=https://api.test.com/\`).
```
[the remark this answers is not in the document]
Comment (ob-pins) [reply apps/api-testing/README.md]: fixed

File: apps/api/src/server/app.js [hunk 2]
```diff
@@ -41,6 +42,10 @@ app.use(
 	})
 );
 
+app.all('*', (req, res, next) => {
+	next(new BackOfficeAppError(`Method is not allowed`, 405));
+});
```
[the remark this answers is not in the document]
Comment (ob-pins) [reply apps/api/src/server/app.js:47]: discussed

File: apps/api-testing/tests/search/search.test.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	blankQueryError
+} from './index.js';
+
+describe('Applications - Search', () => {
+	beforeEach(() => {
+		request.reset();
+	});
+
+	describe('Positive', () => {
+		it('should allow inspectors to perform a search', async () => {
```
[the remark this answers is not in the document]
Comment (ob-pins) [reply apps/api-testing/tests/search/search.test.js:19]: added

**Review by LetsMakeItTechnical — APPROVED** [review]

**Pull request closed without merging**